src/GeniusYield/Providers/Maestro.hs

```diff
@@ -89,6 +95,59 @@ maestroSubmitTx env tx = do
     handleMaestroSubmitError :: Either Maestro.MaestroError a -> IO a
     handleMaestroSubmitError = either (throwIO . SubmitTxException . Text.pack . show . silenceHeadersMaestroClientError) pure
 
+-------------------------------------------------------------------------------
+-- Await tx confirmation
+-------------------------------------------------------------------------------
+
+-- | Awaits for the confirmation of a given 'GYTxId'
+maestroAwaitTxConfirmed :: Maestro.MaestroEnv 'Maestro.V1 -> GYAwaitTx
+maestroAwaitTxConfirmed env p@GYAwaitTxParameters{..} txId = mspvAwaitTx 0
+  where
+    mspvAwaitTx :: Int -> IO ()
+    mspvAwaitTx attempt | maxAttempts <= attempt = throwIO $ GYAwaitTxException p
+    mspvAwaitTx attempt = do
+        eTxInfo <- maestroQueryTx env txId
+        case eTxInfo of
+            Left Maestro.MaestroNotFound -> threadDelay checkInterval >>
+                                            mspvAwaitTx (attempt + 1)
+            Left err -> throwMspvApiError "AwaitTx" err
+            Right txInfo -> msvpAwaitBlock attempt $
+                            Maestro._txDetailsBlockHash $
+                            Maestro.getTimestampedData txInfo
+
+    msvpAwaitBlock :: Int -> Maestro.BlockHash -> IO ()
+    msvpAwaitBlock attempt _ | maxAttempts <= attempt = throwIO $ GYAwaitTxException p
+    msvpAwaitBlock attempt blockHash = do
+        eBlockInfo <- maestroQueryBlock env blockHash
+        case eBlockInfo of
+            Left Maestro.MaestroNotFound -> threadDelay checkInterval >>
```

Comment:
Could this also be the case of `throwMspvApiError`? I mean, is it possible for maestro to have tx info without corresponding block info?

Reply:
This is a good question. I mean, I just wanted to be robust, but your point is valid. Do you think we should remove this case?

Reply:
No, I think it's better the way it is, as we are anyway incrementing the attempt in case block is not found, this certainly feels more robust.